src/samples/Authorization/ManualSignIn/AuthAgent.cs

```diff
@@ -0,0 +1,217 @@
+    /// This agent will handle the sign-in and sign-out processes for a user.
+    /// </summary>
+    /// <param name="options">AgentApplication Configuration objects to configure and setup the Agent Application</param>
+    public AuthAgent(AgentApplicationOptions options) : base(options)
+    {
+        /*
+         During setup of the Agent Application, Register Event Handlers for the Agent. 
+         For this example we will register a welcome message for the user when they join the conversation, then configure sign-in and sign-out commands.
+         Additionally, we will add events to handle notifications of sign-in success and failure,  these notifications will report the local log instead of back to the calling agent. .
+
+         This handler should only register events and setup state as it can be called multiple times before agent events are invoked. 
+        */
+
+        // this sets the default signin handler to the default handler name.  This is used to identify the handler that will be used for the sign-in process later in this code. 
+        _signInHandlerName = Authorization.DefaultHandlerName;
+
+        /*
+         When a conversation update event is triggered. 
+        */
+        OnConversationUpdate(ConversationUpdateEvents.MembersAdded, WelcomeMessageAsync);
+
+        ///*
+        // Handles the user sending a Login or LogOut command using the specific keywords '-signin' and '-signout'
+        //*/
+        // Handles the user sending a Login command using the specific keywords '-signin'
+        OnMessage("-signin", SignInAsync);
+
+        // Handles the user sending a Logout command using the specific keywords '-signout'
+        OnMessage("-signout", async (turnContext, turnState, cancellationToken) =>
+        {
+            // force a user signout to reset the user state
+            // this is needed to reset the token in Azure Bot Services if needed. 
+            await Authorization.SignOutUserAsync(turnContext, turnState, cancellationToken: cancellationToken);
+            await turnContext.SendActivityAsync("You have signed out", cancellationToken: cancellationToken);
+        }, rank: RouteRank.Last);
+
+
+        /*
+         The Authorization Class provides methods and properties to manage and access user authentication tokens
+         You can use this class to interact with the authentication process, including signing in and signing out users, accessing tokens, and handling authentication events.
+         */
+
+        // Register Events for SignIn and SignOut on the authentication class to track the status of the user, notify the user of the status of their authentication process, or log the status of the authentication process.
+        Authorization.OnUserSignInSuccess(OnUserSignInSuccess);
+
+        Authorization.OnUserSignInFailure(async (turnContext, turnState, handlerName, response, initiatingActivity, cancellationToken) =>
```

Comment:
this reads weird, why not create a `OnUserSignInFailure` for symetry with the line above?

Reply:
mainly for brevity, the method pretty much just does a notification.